### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by faust64** [operator-framework/operator-sdk#3857]
#### What did you do?

Leaving my operator running, I would see its memory usage slowly growing.

#### What did you expect to see?

Somewhat stable memory usage. Knowing that eventually, it would stop eating all my memory.

#### What did you see instead? Under which circumstances?

Running an operator over a week, It's quite obvious we have a memory leak somewhere.

![ansible-operator-1 0-memory-leak](https://user-images.githubusercontent.com/211675/92352346-2dcc6f80-f0de-11ea-826f-36165913bc25.png)

operator sources: https://gitlab.com/synacksynack/opsperator/docker-operator
the dashboard I'm using here: https://gitlab.com/synacksynack/opsperator/docker-operator/-/blob/master/roles/prometheus/files/dashboards/Ansible_Operator.json

#### Environment

**Operator type:**

/language ansible

**Kubernetes cluster type:**

vanilla, v1.18.3

`$ operator-sdk version`

First tried reporting this last week:

```
operator-sdk version: "v0.19.2", commit: "4282ce9acdef6d7a1e9f90832db4dc5a212ae850", kubernetes version: "v1.18.2", go version: "go1.13.10 linux/amd64"
```

Now using :v1.0.0 base image building my operator (see Dockerfile in repo mentioned above).
I'ld rarely use the operator-sdk binary.

`$ kubectl version`

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by camilamacedo86** [comment]
Hi @faust64, 

could you please let us know if face the issue with the image`quay.io/operator-framework/ansible-operator:v1.0.0`  as well? also, I see that you made customizations on it [here](https://gitlab.com/synacksynack/opsperator/docker-operator/-/blob/master/Dockerfile#L20-47). See the issue faced in the past: https://github.com/operator-framework/operator-sdk/issues/2112.

So, it might be related to the python and deps added on it.

**Comment by faust64** [comment]
Hi,

I am using the v1.0.0 image - as my Dockerfile would confirm. I haven't used the operator-sdk CLI in ages, don't trust its version, ...
The graph I'm providing shows memory usage during my first week of running v1.0.0.

Regards.

**Comment by camilamacedo86** [comment]
Hi @faust64, 

As described above see that you made customizations on the [Dockerfile](https://gitlab.com/synacksynack/opsperator/docker-operator/-/blob/master/Dockerfile#L20-47) and is not using the same versions of python and etc. See the issue faced in the past: https://github.com/operator-framework/operator-sdk/issues/2112. 

Shows that in the past this problem was solved by using `openshift~=0.10.0`. See the implementation to build a downstream image: https://github.com/openshift/ocp-release-operator-sdk/blob/master/release/ansible/upstream.Dockerfile#L23-L37 
A possible solution shows you ensure that your customizations are using these dependencies. 

Are you able to test it with your customizations and let us know? Are you able to make your project use: the above dependencies and let us know?

**Comment by faust64** [comment]
Hi,

Note I don't pass any environment variable building my image: we're not going through any `if`.
I am only installing openssh and openssl:

```
$ docker run --entrypoint /bin/sh quay.io/operator-framework/ansible-operator:v1.0.0 -c 'rpm -qa'  | sort >rpm-orig
$ kubectl exec -n operator deploy/operator -- rpm -qa | sort >rpm-mine
$ docker run --entrypoint /bin/sh quay.io/operator-framework/ansible-operator:v1.0.0 -c 'pip3 list'  | sort >pip-orig
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
$ kubectl exec -n operator deploy/operator -- pip3 list | sort >pip-mine
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
$
$
$
$ diff rpm-orig rpm-mine 
33a34,35
> fipscheck-1.5.0-4.el8.x86_64
> fipscheck-lib-1.5.0-4.el8.x86_64
72a75
> libedit-3.1-23.20170329cvs.el8.x86_64
121a125,127
> openssh-8.0p1-4.el8_1.x86_64
> openssh-clients-8.0p1-4.el8_1.x86_64
> openssl-1.1.1c-15.el8.x86_64
$ diff pip-orig pip-mine 
$ wc -l pip-mine rpm-mine 
  52 pip-mine
 190 rpm-mine
 242 total
$ grep -E '(ipaddress|ansible|openshift|jmespath)' pip-mine 
ansible (2.9.11)
ansible-runner (1.3.4)
ansible-runner-http (1.0.0)
ipaddress (1.0.23)
jmespath (0.10.0)
openshift (0.10.3)
```
